src/applications/discharge-wizard/components/v2/resultsComponents/StepThree.jsx

```diff
@@ -0,0 +1,240 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  SHORT_NAME_MAP,
+  RESPONSES,
+} from '../../../constants/question-data-map';
+import { BCNR, BCMR, DRB, AFDRB } from '../../../constants';
+import {
+  determineBoardObj,
+  determineBranchOfService,
+  determineVenueAddress,
+} from '../../../helpers';
+
+const StepThree = ({ formResponses }) => {
+  const reason = formResponses[SHORT_NAME_MAP.REASON];
+  const noPrevApp =
+    formResponses[SHORT_NAME_MAP.PREV_APPLICATION] === RESPONSES.NO;
+  const prevAppType = formResponses[SHORT_NAME_MAP.PREV_APPLICATION_TYPE];
+  const prevAppYear = formResponses[SHORT_NAME_MAP.PREV_APPLICATION_YEAR];
+  const dischargeYear = formResponses[SHORT_NAME_MAP.DISCHARGE_YEAR];
+  const dischargeMonth = formResponses[SHORT_NAME_MAP.DISCHARGE_MONTH] || 1;
+  const oldDischarge =
+    new Date().getFullYear() -
+      new Date(dischargeYear, dischargeMonth).getFullYear() >
+    15;
+
+  const boardToSubmit = determineBoardObj(formResponses);
+  const branchOfService = determineBranchOfService(
+    formResponses[SHORT_NAME_MAP.SERVICE_BRANCH],
+  );
+
+  let boardExplanation;
+  let onlineSubmissionMsg;
+
+  if (reason === RESPONSES.REASON_DD215_UPDATE_TO_DD214) {
```

Comment:
This `if` block is pretty unwieldy. It may help if we do some destructuring (pulling variables off `RESPONSES` and `SHORT_NAME_MAP` and putting some of the strings into a constant or function in the constants file. Remember you can always create a function with variables too, e.g.:

```
const makeBoardExplanationForBCMRandBCNR = (boardToSubmitName, boardToSubmitAbbr) => {
  return `the ${boardToSubmit.name}. The ${
        boardToSubmit.abbr
      } was the Board that granted your previous upgrade request, so you must apply to them for a new DD214.`;
};
```

Reply:
Thanks, I agree I'll do some clean up.